python/ray/ml/checkpoint.py

```diff
@@ -291,27 +298,32 @@ def from_directory(cls, path: str) -> "Checkpoint":
         Returns:
             Checkpoint: checkpoint object.
         """
-        return Checkpoint(local_path=path)
-
-    def to_directory(self, path: Optional[str] = None, dedup: bool = True) -> str:
-        """Write checkpoint data to directory.
-
-        Args:
-            path: Target directory to restore data in. If not specified,
-
-        Returns:
-            str: Directory containing checkpoint data.
-        """
-        path = path if path is not None else _temporary_checkpoint_dir()
-
-        os.makedirs(path, exist_ok=True)
-        # Drop marker
-        open(os.path.join(path, ".is_checkpoint"), "a").close()
+        return cls(local_path=path)
 
+    def _get_temporary_checkpoint_dir(self) -> str:
+        """Return the name for the temporary checkpoint dir."""
+        if self._obj_ref:
```

Comment:
Should we do this also for URIs? (just hash the URI). We can do this in a follow-up